aws-predictions/src/main/java/com/amplifyframework/predictions/aws/service/AWSPollyService.java

```diff
@@ -0,0 +1,116 @@
+import com.amplifyframework.predictions.result.TextToSpeechResult;
+import com.amplifyframework.util.UserAgent;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.services.polly.AmazonPollyClient;
+import com.amazonaws.services.polly.AmazonPollyPresigningClient;
+import com.amazonaws.services.polly.model.OutputFormat;
+import com.amazonaws.services.polly.model.SynthesizeSpeechRequest;
+import com.amazonaws.services.polly.model.SynthesizeSpeechResult;
+import com.amazonaws.services.polly.model.TextType;
+
+import java.io.InputStream;
+
+/**
+ * Predictions service for performing text to speech conversion.
+ */
+final class AWSPollyService {
+    private static final int MP3_SAMPLE_RATE = 24_000;
+
+    private final AmazonPollyClient polly;
+    private final AWSPredictionsPluginConfiguration pluginConfiguration;
+
+    AWSPollyService(AWSPredictionsPluginConfiguration pluginConfiguration) {
+        this.polly = createPollyClient();
+        this.pluginConfiguration = pluginConfiguration;
+    }
+
+    private AmazonPollyClient createPollyClient() {
+        AWSCredentialsProvider credentialsProvider = AWSMobileClient.getInstance();
+        ClientConfiguration configuration = new ClientConfiguration();
+        configuration.setUserAgent(UserAgent.string());
+        return new AmazonPollyPresigningClient(credentialsProvider, configuration);
+    }
+
+    void synthesizeSpeech(
+            @NonNull String text,
+            @NonNull AWSVoiceType voiceType,
+            @NonNull Consumer<TextToSpeechResult> onSuccess,
+            @NonNull Consumer<PredictionsException> onError
+    ) {
+        try {
+            InputStream data = synthesizeSpeech(text, voiceType);
+            onSuccess.accept(TextToSpeechResult.fromAudioData(data));
+        } catch (PredictionsException exception) {
+            onError.accept(exception);
+        }
+    }
+
+    private InputStream synthesizeSpeech(String text, AWSVoiceType voiceType) throws PredictionsException {
+        final String languageCode;
+        final String voiceId;
+        if (AWSVoiceType.UNKNOWN.equals(voiceType)) {
+            // Obtain voice + language from plugin configuration by default
+            SpeechGeneratorConfiguration config = pluginConfiguration.getSpeechGeneratorConfiguration();
+            languageCode = config.getLanguage();
+            voiceId = config.getVoice();
+        } else {
+            // Override configuration defaults if explicitly specified in the options
+            languageCode = voiceType.getLanguageCode();
+            voiceId = voiceType.getName();
+        }
+
+        SynthesizeSpeechRequest request = new SynthesizeSpeechRequest()
+                 .withText(text)
+                 .withTextType(TextType.Text)
+                 .withLanguageCode(languageCode)
```

Comment:
Apparently [languageCode is an optional parameter](https://github.com/aws-amplify/aws-sdk-android/blob/09ccb6110232c8103dcb5037f34840f1c2240261/aws-android-sdk-polly/src/main/java/com/amazonaws/services/polly/model/SynthesizeSpeechRequest.java#L47-L68), and if you don't include it, multilingual voices will fall back to a default language.  I guess this is how iOS is able to not include this parameter at all, and still have a mostly working implementation.  

I like that we now have multilingual support for Android.  I'd suggest reaching out to iOS and suggesting that they add similar support.

Reply:
yep! I also noticed this shortly after. It should definitely be supported by both platforms ideally.